### PR TITLE
Fix file arrow scanner column index out of range core.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

```FileArrowScanner::_append_batch_to_block``` may core while executing tpch queries. It also may core while executing query like ```select count(1) from xxx```.

## Problem summary
The `get_column_index(column_name)` may return an index value that out of the `_batch` column range. For example, 
```select count(1) from xxx``` has only 1 column, but `get_column_index(column_name)` may return a value >= 1.

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/arrow/arrow_reader.h [hunk 1]
```diff
@@ -100,6 +100,7 @@ class ArrowReaderWrap {
     int get_cloumn_index(std::string column_name);
 
     void prefetch_batch();
+    bool is_case_sensitive() {return _caseSensitive; }
```
Comment (morningman) [inline be/src/exec/arrow/arrow_reader.h:103]:
```suggestion
    bool is_case_sensitive() { return _caseSensitive; }
```

Comment (Gabriel39) [inline be/src/exec/arrow/arrow_reader.h:103]: How about to use `_case_sensitive` instead?

Comment (Jibing-Li) [reply be/src/exec/arrow/arrow_reader.h:103]: I'll fix it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**